odc/geo/cog/_az.py

```diff
@@ -0,0 +1,191 @@
+import base64
+from typing import Any, Union
+
+from azure.storage.blob import BlobBlock, BlobServiceClient
```

Comment:
move this import into methods that need that (and tell pylint it's ok). We need to be able to import from here even when `azure` is not installed in the environment.

Reply:
I tried, but I couldn't get `python -m unittest tests/test_az.py` to behave without the import being outside of the class. Would you know how to do that?